### PR TITLE
Remove asserts from geomstats' main files

### PR DESCRIPTION
**Pull request by ninamiolane**
This PR addresses the DeepSource error "Assert statement used outside of tests", by transforming `asserts` into errors, see issue #501.

By doing so, it also starts to address issue #538 and creates a file to uniformize the error messages throughout the code base.

### CODE REVIEW

File: geomstats/geometry/matrices.py [hunk 1]
```diff
@@ -178,14 +180,10 @@ def __init__(self, m, n):
...
         tangent_vec_a = gs.to_ndarray(tangent_vec_a, to_ndim=3)
-        n_tangent_vecs_a, _, _ = tangent_vec_a.shape
-
         tangent_vec_b = gs.to_ndarray(tangent_vec_b, to_ndim=3)
-        n_tangent_vecs_b, _, _ = tangent_vec_b.shape
-
-        assert n_tangent_vecs_a == n_tangent_vecs_b
 
-        inner_prod = gs.einsum("nij,nij->n", tangent_vec_a, tangent_vec_b)
+        inner_prod = gs.einsum(
```
Comment (nguigs) [inline geomstats/geometry/matrices.py:185]: :heart_eyes:

File: geomstats/geometry/product_riemannian_metric.py [hunk 2]
```diff
@@ -98,7 +98,9 @@ def is_intrinsic(self, point):
         intrinsic: bool
             Whether intrinsic coordinates are used for all manifolds.
         """
-        assert self.default_point_type == 'vector'
+        if self.default_point_type != 'vector':
```
Comment (nguigs) [inline geomstats/geometry/product_riemannian_metric.py:101]: in fact if we reach that point there really is a problem, it's not that it was not implemented

Comment (ninamiolane) [reply geomstats/geometry/product_riemannian_metric.py:101]: Modified!

**Comment by ninamiolane** [comment]
@nguigs thanks for looking at it! Yes, code on default_point_type could be factorized, and actually I was thinking about it in the context of code vectorization. Because so far, we have 3 mechanisms that have to do with vectorization:
- decorator,
- einsums,
- point_types,
and it could be great to converge to the minimal one, both in terms of code efficiency and code readability.

I've put a TODO for now.

**Pull request merged**